Vingette for step by step guide how to perform t-test. (#286)
Olink-Proteomics/OlinkRPackage: OlinkAnalyze/vignettes/study_design_mixedmodel.Rmd

---

> title: "A Guide to Comparative Analysis of logitudinal data across muliple groups"

Reviewer: ```suggestion
title: "A Guide to Comparative Analysis of longitudinal data across multiple groups"
```

author: "Olink DS team"
date: "`r Sys.Date()`"
output: 
  html_vignette:
    latex_engine: xelatex
    toc: true
    toc_depth: 2
vignette: >

>   %\VignetteIndexEntry{Olink® Analyze Vignette}

Reviewer: ```suggestion
  %\VignetteIndexEntry{A Guide to Comparative Analysis of longitudinal data across multiple groups}
```

  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
--- 

```{r setup, include = FALSE}
knitr::opts_chunk$set(
collapse = TRUE,
comment = "#>")
options(tibble.print_min = 4L, tibble.print_max = 4L)
library(dplyr)
library(ggplot2)
library(stringr)
library(OlinkAnalyze)

df <- OlinkAnalyze::npx_data1 %>% 
  filter(!str_detect(SampleID, 'CONTROL'))
df_QC <- OlinkAnalyze::npx_data1 %>% 
  filter(!str_detect(SampleID, 'CONTROL'))

```

# Introduction  

> This vignette is designed to assist users working with longitudinal samples divided across multiple groups using linear mixed models. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing Olink data when the samples are measured over time in different groups using the Olink R package. An example can be samples measured over time where half the subjects have been revived a treatment and the other half have not. 

Reviewer: ```suggestion
This vignette is designed to assist users working with longitudinal samples divided across multiple groups using linear mixed models. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing Olink data when the samples are measured over time in different groups using Olink's R package, Olink Analyze. An example can be samples measured over time where half the subjects have been revived a treatment and the other half have not. 
```

# Installation 

You can install Olink® Analyze from CRAN which is needed for the following walkthrough.

```{r, eval=FALSE}
install.packages("OlinkAnalyze")
```

After installing the package, the package needs to be loaded. Several packages from the tidyverse suite of packages are also used throughout this vignette, such as dplyr and stringr. Each individual package can be loaded using the _library_ function (`library("package")`) or the entire tidyverse suite of packages can be loaded as shown below.

```{r, eval=FALSE}
library(OlinkAnalyze)
library(tidyverse)
```

# Study overview

This vignette will help you if you have Olink data and are unsure on how to compare your longitudinal samples spread across multiple groups.  The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R use the following code:

```{r message=FALSE, eval=FALSE}
data <- read_NPX("~/NPX_file_location.csv")

> # Argument should be the location of your NPX data. Data are usually in parquet, csv, xlsx or zip format depending on the product.

Reviewer: ```suggestion
# Argument should be the location of your NPX data. 
# Data are usually in parquet, csv, xlsx or zip format.
# Data file format depends on the product.
```

```  

Sometime including outlier or samples which don't pass QC can still be reasonable, especially in cases with very few samples.

# Analysis 
## Linear mixed model

Linear mixed model (LMM) is used to analyze data that have both fixed effects and random effects. It is particularly useful when dealing with data that exhibit hierarchical or nested structures or repeated measures. LMM extends the concept of a simple linear regression model to account for these complexities.

In an LMM, the "linear" aspect refers to the fact that the relationships between the dependent variable and the independent variables are assumed to be linear. The "mixed" aspect refers to the inclusion of both fixed effects and random effects in the model.

Here we will use the example data provided within Olink Analyze to assess differences between groups using the Treatment and Time variable. Six distinct groups, Treated and Untreated in 3 timepoints exist in this data, as shown below.

```{r message=FALSE, eval=T}
df %>%  
  select(SampleID, Treatment, Time, Subject) %>% 
  unique() %>%
  select(Treatment, Time) %>% 
  table

```  

> LMM assesses the null hypothesis that no significant effect of the independent variables on the dependent variable. If the p-value associated with the test is lower than a predetermined significance level (typically 0.05), we can reject the null hypothesis and infer a significant effect of at least one independent variable on the dependent variable. This indicates that the independent variables, included in the model, have a statistically significant impact on explaining the variation in the dependent variable.

Reviewer: ```suggestion
LMM assesses the null hypothesis that there is no significant effect of the independent variables on the dependent variable. If the p-value associated with the test is lower than a predetermined significance level (typically 0.05), we can reject the null hypothesis and infer a significant effect of at least one independent variable on the dependent variable. This indicates that the independent variables, included in the model, have a statistically significant impact on explaining the variation in the dependent variable.
```

Conversely, if the p-value exceeds the significance level, we do not have enough evidence to reject the null hypothesis, suggesting an absence of a statistically significant effect of the independent variables on the dependent variable in the context of the linear mixed model.

> The LMM also take into account the random effect, which is a critical component of the model. Random effects are used to model the variability within groups or clusters in the data, accounting for correlations and hierarchical structures that might exist. These random effects are essential in capturing the dependencies and correlations in the data that cannot be explained solely by the independent variables. The random effect are usually subject for Olink studies. 

Reviewer: ```suggestion
The LMM also take into account the random effect, which is a critical component of the model. Random effects are used to model the variability within groups or clusters in the data, accounting for correlations and hierarchical structures that might exist. These random effects are essential in capturing the dependencies and correlations in the data that cannot be explained solely by the independent variables. The random effect is usually within subject correlation for Olink studies. 
```

```{r message=FALSE, eval=T}
#the post hoc test should only be performed on proteins which have a significant result in the ANOVA
significant_proteins_Time <- lmer_results %>% 
  filter(Threshold == 'Significant') %>% 
  filter(term == 'Time') %>% 
  pull(OlinkID)

lmer_posthoc_results_Time<-olink_lmer_posthoc(df,
                    olinkid_list = significant_proteins_Time, #significant proteins
                    variable=c("Time", 'Treatment'),
                    effect = 'Time',
                    random = 'Subject')

```  

The *olink_lmer_posthoc* function arrange the results based on adjusted p-value. For example in **Table 4**  on the first row the assay CCL18 have a significant difference between Baseline and week.6 of 2.67 NPX. One posthoc test should be for each significant assay in said term.  

```{r message=FALSE, eval=T, echo=FALSE}
lmer_posthoc_results_Time %>%
  head(5)  %>% 
  select(Assay, OlinkID, UniProt, Panel, contrast ,estimate, Adjusted_pval, Threshold) %>% 
  knitr::kable(format = "html",
               booktabs = T,
               linesep = "",
               digits = 4,
               longtable = T,
               row.names = F,

>                caption = "**Table 4.** ANOVA post hoc results of the 5 most significant contrasts and proteins. Some coloumns have been left out due to space.",

Reviewer: ```suggestion
               caption = "**Table 4.** ANOVA post hoc results of the 5 most significant contrasts and proteins. Some columns have been left out due to space.",
```

               label = "manifest")
```  

## Plots

> To illustrate the results from a LMM a point range plot are a common way. These visualizations provide valuable insights into the differences between the groups and help in understanding the significance and effect sizes of the observed effects.

Reviewer: ```suggestion
A point range plot is a common way to illustrate the results from an LMM. These visualizations provide valuable insights into the differences between the groups and help in understanding the significance and effect sizes of the observed effects.
```

To illustrate the point range plot the *olink_lmer_plot* function can be used. In the example below only the top 6 significant proteins are plotted. 

```{r message=FALSE, eval=T, fig.height = 4, fig.width = 8, fig.align = "center"}
top6Significant_time<-lmer_results %>% 
  filter(Threshold == 'Significant') %>% 
  filter(term == 'Time') %>% 
  head(6) %>%   # just the top 6 for visualization
  pull(OlinkID) # pull out the OlinkID which is needed in olink_boxplot

lmer_top6_time<-olink_lmer_plot(df = df,
              variable=c("Time", 'Treatment'),
              random = 'Subject',
              x_axis_variable = 'Time',
              col_variable = 'Treatment',
              olinkid_list = top6Significant_time) 

lmer_top6_time
``` 

# Saving results

Having the results saved in csv files (or other format) and the plots in pdf (or other format) can making the results easier to work with, especially if the results are being shared with other people that do not have access to R. 

The following code can be used to save the results as a csv file.

```{r message=FALSE, eval=F}

write.table(lmer_results,

>             paste0('LMER_Results.csv'),

Reviewer: same question as before with paste0

            sep=';',row.names = F, quote = F)

> write.table(lmer_posthoc_results_Time, # each term should be saved

Reviewer: same question with paste0

            paste0('LMER_posthoc_Time_Results.csv'),
            sep=';',row.names = F, quote = F)

> ```  

Reviewer: Add contact us section
